SourceX/controls/plrctrls.cpp

```diff
@@ -664,6 +671,81 @@ void PerformPrimaryAction()
 	Interact();
 }
 
+bool SpellHasActorTarget()
+{
+	int v = plr[myplr]._pRSpell;
+	if (v == SPL_INVALID || v == SPL_TOWN || v == SPL_TELEPORT)
+		return false;
+
+	if (spelldata[v].sTargeted || v == SPL_BONESPIRIT || v == SPL_FLARE) {
+		if (!spelldata[v].sTownSpell && pcursmonst != -1) {
+			cursmx = monster[pcursmonst]._mx;
+			cursmy = monster[pcursmonst]._my;
+			return true;
+		}
+		if ((spelldata[v].sTownSpell || !FriendlyMode) && pcursplr != -1) {
+			cursmx = plr[pcursplr]._px;
+			cursmy = plr[pcursplr]._py;
+			return true;
+		}
+	}
+
+	return false;
+}
+
+void UpdateSpellTarget()
+{
+	if (SpellHasActorTarget())
+		return;
+
+	pcursplr = -1;
+	pcursmonst = -1;
+
+	cursmx = plr[myplr]._px;
+	cursmy = plr[myplr]._py;
+	switch (plr[myplr]._pdir) {

```

Comment:
Perhaps it would be cleaner to make a lookup table

Reply:
Would you make a table for each axis?

Reply:
Perhaps one 2-dimensional table to fetch both values at once?

Something like this:

```c++
static const int kOffsets[8][2] = {
  {  1,  1 }, // DIR_S
  {  0,  1 }, // DIR_SW
  { -1,  1 }, // DIR_W
  { -1,  0 }, // DIR_NW
  { -1, -1 }, // DIR_N
  {  0, -1 }, // DIR_NE
  {  1, -1 }, // DIR_E
  {  1,  0 }, // DIR_SE
};
const auto &offset = kOffsets[plr[myplr]._pdir];
cursmx = plr[myplr]._px + offset[0];
cursmy = plr[myplr]._py + offset[1];
```